File: Data Structures & Algorithms/duplicate-integer/submission-10.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        seen = {}

        for n in nums:
            if seen.get(n, 0) >= 1:
                 return True

            seen.setdefault(n, seen.get(n, 0) + 1)

        return False;

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-10.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const seen = new Set(nums);
        let longest = 0;

        for (let n of nums) {
            if(!seen.has(n - 1)) {
                let current = 1;

                while(seen.has(n + current)) {
                    current++;
                }

                longest = Math.max(longest, current);
            }
        }

        return longest

    }
}
